lib/checkexceptionsafety.cpp

```diff
@@ -251,6 +259,19 @@ static const Token * functionThrowsRecursive(const Function * function, std::set
 
     for (const Token *tok = function->functionScope->bodyStart->next();
          tok != function->functionScope->bodyEnd; tok = tok->next()) {
+        if (Token::simpleMatch(tok, "if")) {
+            tok = tok->astParent();
+            const Token *condTok = tok->astOperand2();
+            // check if condition is possibly false, since template arguments are not set to known
+            if (isPossibleBool(condTok, false)) {
```

Comment:
Somehow we should check if the possible values are template arguments. You can check `Token::templateArg` for  "possible" operands .

If the condition is false you skip the if-body but if the condition is true you do nothing, you should skip the `else` body then.

Reply:
The utility function I introduce in #7045 would be useful for this, so maybe I should wait for that one.